ckanext/s3filestore/uploader.py

```diff
@@ -188,19 +195,28 @@ def get_signed_url_to_key(self, key, extra_params={}):
         # check whether the object exists in S3
         metadata = client.head_object(Bucket=self.bucket_name, Key=key)
 
+        # check whether the object is publicly readable
+        acl = client.get_object_acl(Bucket=self.bucket_name, Key=key)
+        is_public_read = any(
+            grant['Grantee']['Type'] == 'Group'
+            and grant['Grantee'].get('URI', '').endswith('AllUsers')
```

Comment:
unsure if AllUsers is always at end of acl list, should this be contains? 
Also we may want to check for the super override for public visability on the bucket level (outside of acl)
This should be cached for 5min intervals (at least) to reduce api calls (IgnorePublicAcls, RestrictPublicBuckets, s3:GetBucketPublicAccessBlock, s3:GetAccountPublicAccessBlock)
https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-control-block-public-access.html

Reply:
As far as I've seen, the 'public-read' grant always uses `http://acs.amazonaws.com/groups/global/AllUsers`; that's what the console interface provides, and that's what you get if you specify the 'public-read' ACL when creating an object.

Reply:
in dev turn on full block via aws console and see what happens on acl calls, (if public read is still visible). My guess is that acl will still turn back what is set, but when accessing file, it will be blocked.